File: easydoc-javadoc/src/main/java/com/star/easydoc/service/variable/VariableGenerator.java
package com.star.easydoc.service.variable;

import com.intellij.psi.PsiElement;

/**
 * 变量生成器
 *
 * @author wangchao
 * @date 2019/12/07
 */
public interface VariableGenerator {
    /**
     * 生成
     *
     * @param element 元素
     * @return {@link String}
     */
    String generate(PsiElement element);
}

File: easydoc-kdoc/src/main/kotlin/com/star/easydoc/kdoc/service/variable/VariableGenerator.kt
package com.star.easydoc.kdoc.service.variable

import com.intellij.psi.PsiElement
import com.star.easydoc.common.config.EasyDocConfig

/**
 * 变量生成器
 *
 * @author wangchao
 * @date 2019/12/07
 */
interface VariableGenerator {
    /**
     * 生成
     *
     * @param element 元素
     * @return [String]
     */
    fun generate(element: PsiElement): String

    /**
     * 获取配置
     *
     * @return [EasyDocConfig]
     */
    val config: EasyDocConfig
}